lib/puppet/provider/pulp_rpmbind/consumer.rb

```diff
@@ -0,0 +1,41 @@
+Puppet::Type.type(:pulp_rpmbind).provide(:consumer) do
+  desc 'Bind/unbind to an RPM repo'
+
+  confine osfamily: :redhat
+  commands consumer: '/bin/pulp-consumer'
+  commands grep:     '/bin/grep'
+
+  def self.instances
+    begin
+      binds = grep('-oP', '^\[\K.*(?=\]$)', '/etc/yum.repos.d/pulp.repo')
```

Comment:
I'm leaning to going with grep for now but create an issue for pulp-consumer to list repos.